flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamMultipleInputProcessor.java

```diff
@@ -47,6 +49,26 @@ public StreamMultipleInputProcessor(
             StreamOneInputProcessor<?>[] inputProcessors) {
         this.inputSelectionHandler = inputSelectionHandler;
         this.inputProcessors = inputProcessors;
+        this.availabilityHelper =
+                MultipleInputAvailabilityHelper.newInstance(inputProcessors.length);
+        this.availabilityHelper.init();
+    }
+
+    @Override
+    public boolean isAvailable() {
+        if (inputSelectionHandler.isAnyInputAvailable()
+                || inputSelectionHandler.areAllInputsFinished()) {
+            return true;
+        } else {
+            boolean isAvailable = false;
+            for (int i = 0; i < inputProcessors.length; i++) {
+                isAvailable =
+                        !inputSelectionHandler.isInputFinished(i)
+                                && inputSelectionHandler.isInputSelected(i)
+                                && inputProcessors[i].isAvailable();
+            }
+            return isAvailable;
```

Comment:
The same question as below. Isn't this whole loop a duplicated check from the `inputSelectionHandler.isAnyInputAvailable()` condition? In other words, couldn't we just simply `return false` in this `else` branch?